packages/core/src/server.ts

```diff
@@ -21,3 +21,10 @@ export interface Server {
   start(): Promise<void>;
   stop(): Promise<void>;
 }
+
+export type NamedServer = {
+  name: string;
+  // tslint:disable-next-line:no-any
+  ctor: Constructor<any>;
+  [key: string]: {};
```

Comment:
What is the purpose of this? If you want to support arbitrary extensions, then I think the value should be `any` (not `{}`).

Reply:
Ugh, sorry. My TSLint has auto-fix on save so if I forget to mark an exception to this rule, it replaces it with {}.

As a side note, do we really want this rule in place? We seem to be creating tons of exemptions throughout our code.

Reply:
The problem with `any` is that once it gets into your types, it will start propagating through related code. When TypeScript sees `any`, it basically disables entire type checking. Because `any` propagates, it's very often easy to miss the places where you are using `any` and thus loosing type safety. The exceptions are ugly, but at least they are forcing us to think twice about using `any`. But I am fine to discuss whether we should be keeping this rule.